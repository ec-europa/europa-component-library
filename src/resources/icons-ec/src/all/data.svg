<svg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24 18C13.4442 18 8 13.7941 8 12C8 10.2059 13.4442 6 24 6C34.5567 6 40 10.2059 40 12C40 13.7941 34.5567 18 24 18ZM40 24C40 26.1252 34.5567 30 24 30C13.4442 30 8 26.1252 8 24V18C11.7315 20.2539 17.5103 21.7091 24 22C30.4916 21.7091 36.2685 20.2539 40 18V24ZM40 36C40 38.1255 34.5567 42 24 42C13.4442 42 8 38.1255 8 36V30C11.7315 32.2538 17.5103 33.7089 24 34C30.4916 33.7089 36.2685 32.2538 40 30V36ZM24 2C12.9713 2 4 6.31954 4 12V36C4 41.6805 12.9713 46 24 46C35.0287 46 44 41.6805 44 36V12C44 6.31954 35.0287 2 24 2Z" fill="black"/>
<mask id="mask0_1_1085491" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="4" y="2" width="40" height="44">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24 18C13.4442 18 8 13.7941 8 12C8 10.2059 13.4442 6 24 6C34.5567 6 40 10.2059 40 12C40 13.7941 34.5567 18 24 18ZM40 24C40 26.1252 34.5567 30 24 30C13.4442 30 8 26.1252 8 24V18C11.7315 20.2539 17.5103 21.7091 24 22C30.4916 21.7091 36.2685 20.2539 40 18V24ZM40 36C40 38.1255 34.5567 42 24 42C13.4442 42 8 38.1255 8 36V30C11.7315 32.2538 17.5103 33.7089 24 34C30.4916 33.7089 36.2685 32.2538 40 30V36ZM24 2C12.9713 2 4 6.31954 4 12V36C4 41.6805 12.9713 46 24 46C35.0287 46 44 41.6805 44 36V12C44 6.31954 35.0287 2 24 2Z" fill="white"/>
</mask>
<g mask="url(#mask0_1_1085491)">
</g>
</svg>
