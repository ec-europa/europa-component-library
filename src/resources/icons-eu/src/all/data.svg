<?xml version="1.0" encoding="UTF-8" ?>
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
  width="24pt"
  height="24pt"
  viewBox="0 0 24 24"
  version="1.1"
>
<g id="surface1">
<path
      d="M 12 9.25 C 7.160156 9.25 4.667969 7.324219 4.667969 6.5 C 4.667969 5.675781 7.160156 3.75 12 3.75 C 16.839844 3.75 19.332031 5.675781 19.332031 6.5 C 19.332031 7.324219 16.839844 9.25 12 9.25 Z M 19.332031 12 C 19.332031 12.972656 16.839844 14.75 12 14.75 C 7.160156 14.75 4.667969 12.972656 4.667969 12 L 4.667969 9.25 C 6.375 10.28125 9.027344 10.949219 12 11.082031 C 14.976562 10.949219 17.625 10.28125 19.332031 9.25 Z M 19.332031 17.5 C 19.332031 18.472656 16.839844 20.25 12 20.25 C 7.160156 20.25 4.667969 18.472656 4.667969 17.5 L 4.667969 14.75 C 6.375 15.78125 9.027344 16.449219 12 16.582031 C 14.976562 16.449219 17.625 15.78125 19.332031 14.75 Z M 12 1.917969 C 6.945312 1.917969 2.832031 3.894531 2.832031 6.5 L 2.832031 17.5 C 2.832031 20.105469 6.945312 22.082031 12 22.082031 C 17.054688 22.082031 21.167969 20.105469 21.167969 17.5 L 21.167969 6.5 C 21.167969 3.894531 17.054688 1.917969 12 1.917969 Z M 12 1.917969 "
    />
</g>
</svg>
