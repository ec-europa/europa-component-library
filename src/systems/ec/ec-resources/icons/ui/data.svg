<!-- Generated by IcoMoon.io -->
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="28" height="32" viewBox="0 0 28 32">
<title>data</title>
<path d="M14.075 0.376c-7.791 0-14.075 3.078-14.075 7.022v18.115c0 3.848 6.284 7.022 14.075 7.022s14.043-3.11 14.043-6.958v-18.18c0-3.944-6.38-7.022-14.043-7.022zM25.137 25.513c0 1.218-3.751 4.008-11.062 4.008-7.214 0-11.062-2.789-11.062-4.008v-4.713c2.533 1.603 6.541 2.725 11.062 2.725 4.393 0 8.432-1.026 11.062-2.725v4.713zM25.137 16.503c0 1.218-3.751 4.040-11.062 4.040-7.214 0-11.062-2.822-11.062-4.040v-4.681c2.533 1.571 6.541 2.693 11.062 2.693 4.393 0 8.432-1.026 11.062-2.693v4.681zM14.075 11.437c-7.214 0-11.062-2.822-11.062-4.040s3.751-4.008 11.062-4.008c7.214 0 11.062 2.789 11.062 4.008s-3.848 4.040-11.062 4.040z"></path>
</svg>
