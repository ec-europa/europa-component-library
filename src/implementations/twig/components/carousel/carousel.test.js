import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe } from 'vitest-axe';

import data from '@ecl/specs-component-carousel/demo/data';

describe('Carousel', () => {
  const template = '@ecl/carousel/carousel.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, data, true)),
      ).toHaveNoViolations();
    });
  });
});
