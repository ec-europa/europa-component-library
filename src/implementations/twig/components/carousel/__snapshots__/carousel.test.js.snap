// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Default renders correctly 1`] = `
<jest>
  <div
    class="ecl-carousel"
    data-ecl-auto-init="Carousel"
  >
    <div
      class="ecl-carousel__container"
    >
      <div
        class="ecl-carousel__slides"
        id="ecl-carousel-slider"
      >
        <div
          aria-label="slide 1 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-box ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Lorem ipsum dolor sit amet
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          aria-label="slide 2 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-overlay ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Duis vitae pulvinar turpis
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          aria-label="slide 3 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-overlay ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Donec maximus pharetra ex a ultricies
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur. Duis vitae pulvinar turpis. Donec maximus pharetra ex a ultricies
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <button
      class="ecl-button ecl-button--ghost ecl-carousel__prev ecl-button--icon-only"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Previous slides
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-icon--rotate-270 ecl-button__icon"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </span>
    </button>
    <button
      class="ecl-button ecl-button--ghost ecl-carousel__next ecl-button--icon-only"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Next slides
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-button__icon"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-carousel__controls"
    >
      <div
        class="ecl-container"
      >
        <div
          class="ecl-carousel__autoplay"
        >
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__play ecl-button--icon-only"
            type="button"
          >
            <span
              class="ecl-button__container"
            >
              <span
                class="ecl-button__label"
                data-ecl-label="true"
              >
                Play carousel
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--primary ecl-button__icon ecl-carousel__icon-default"
                data-ecl-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#play"
                />
              </svg>
            </span>
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__pause ecl-button--icon-only"
            type="button"
          >
            <span
              class="ecl-button__container"
            >
              <span
                class="ecl-button__label"
                data-ecl-label="true"
              >
                Pause carousel
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--primary ecl-button__icon ecl-carousel__icon-default"
                data-ecl-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#pause"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="ecl-carousel__navigation"
        >
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            1
          
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            2
          
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            3
          
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Carousel Default renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-carousel"
    data-ecl-auto-init="Carousel"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-carousel__container"
    >
      <div
        class="ecl-carousel__slides"
        id="ecl-carousel-slider"
      >
        <div
          aria-label="slide 1 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-box ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Lorem ipsum dolor sit amet
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          aria-label="slide 2 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-overlay ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Duis vitae pulvinar turpis
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          aria-label="slide 3 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-overlay ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Donec maximus pharetra ex a ultricies
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur. Duis vitae pulvinar turpis. Donec maximus pharetra ex a ultricies
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <button
      class="ecl-button ecl-button--ghost ecl-carousel__prev ecl-button--icon-only"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Previous slides
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-icon--rotate-270 ecl-button__icon"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </span>
    </button>
    <button
      class="ecl-button ecl-button--ghost ecl-carousel__next ecl-button--icon-only"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Next slides
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-button__icon"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-carousel__controls"
    >
      <div
        class="ecl-container"
      >
        <div
          class="ecl-carousel__autoplay"
        >
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__play ecl-button--icon-only"
            type="button"
          >
            <span
              class="ecl-button__container"
            >
              <span
                class="ecl-button__label"
                data-ecl-label="true"
              >
                Play carousel
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--primary ecl-button__icon ecl-carousel__icon-default"
                data-ecl-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#play"
                />
              </svg>
            </span>
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__pause ecl-button--icon-only"
            type="button"
          >
            <span
              class="ecl-button__container"
            >
              <span
                class="ecl-button__label"
                data-ecl-label="true"
              >
                Pause carousel
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--primary ecl-button__icon ecl-carousel__icon-default"
                data-ecl-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#pause"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="ecl-carousel__navigation"
        >
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            1
          
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            2
          
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            3
          
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`Carousel Default renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-carousel custom-class custom-class--test"
    data-ecl-auto-init="Carousel"
  >
    <div
      class="ecl-carousel__container"
    >
      <div
        class="ecl-carousel__slides"
        id="ecl-carousel-slider"
      >
        <div
          aria-label="slide 1 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-box ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Lorem ipsum dolor sit amet
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          aria-label="slide 2 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-overlay ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Duis vitae pulvinar turpis
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          aria-label="slide 3 of 3"
          class="ecl-carousel__slide"
          role="group"
        >
          <section
            class="ecl-banner ecl-banner--text-overlay ecl-banner--m"
            data-ecl-auto-init="Banner"
            data-ecl-banner=""
          >
            <div
              class="ecl-banner__picture-container"
            >
              <picture
                class="ecl-picture ecl-banner__picture"
                data-ecl-banner-image=""
              >
                <img
                  alt="Alternative text"
                  class="ecl-banner__image"
                  src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
                />
              </picture>
              <div
                class="ecl-banner__credit"
              >
                © Copyright or credit
              </div>
            </div>
            <div
              class="ecl-container"
            >
              <div
                class="ecl-banner__container"
                data-ecl-banner-container=""
              >
                <div
                  class="ecl-banner__content"
                >
                  <div
                    class="ecl-banner__title"
                  >
                    <span
                      class="ecl-banner__title-text"
                    >
                      Donec maximus pharetra ex a ultricies
                    </span>
                  </div>
                  <p
                    class="ecl-banner__description"
                  >
                    <span
                      class="ecl-banner__description-text"
                    >
                      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur. Duis vitae pulvinar turpis. Donec maximus pharetra ex a ultricies
                    </span>
                  </p>
                  <div
                    class="ecl-banner__cta"
                  >
                    <a
                      class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
                      href="/example"
                    >
                      <span
                        class="ecl-link__label"
                      >
                        Subscribe
                      </span>
                      <svg
                        aria-hidden="true"
                        class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                        focusable="false"
                      >
                        <use
                          xlink:href="/icons.svg#corner-arrow"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <button
      class="ecl-button ecl-button--ghost ecl-carousel__prev ecl-button--icon-only"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Previous slides
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-icon--rotate-270 ecl-button__icon"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </span>
    </button>
    <button
      class="ecl-button ecl-button--ghost ecl-carousel__next ecl-button--icon-only"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Next slides
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-button__icon"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-carousel__controls"
    >
      <div
        class="ecl-container"
      >
        <div
          class="ecl-carousel__autoplay"
        >
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__play ecl-button--icon-only"
            type="button"
          >
            <span
              class="ecl-button__container"
            >
              <span
                class="ecl-button__label"
                data-ecl-label="true"
              >
                Play carousel
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--primary ecl-button__icon ecl-carousel__icon-default"
                data-ecl-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#play"
                />
              </svg>
            </span>
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__pause ecl-button--icon-only"
            type="button"
          >
            <span
              class="ecl-button__container"
            >
              <span
                class="ecl-button__label"
                data-ecl-label="true"
              >
                Pause carousel
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--primary ecl-button__icon ecl-carousel__icon-default"
                data-ecl-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#pause"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="ecl-carousel__navigation"
        >
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            1
          
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            2
          
          </button>
          <button
            class="ecl-button ecl-button--ghost ecl-carousel__navigation-item"
          >
            
            3
          
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
